doc/commits.rst

```diff
@@ -0,0 +1,46 @@
+How to create a good commit history
+===================================
+
+Commit structure
+-----------------
+
+- Focus on a specific change. Don't try to change everything at once.
+  If the plan is to implement a series of things, do them one by one.
+
+- Do not mix changes concerning different features in one commit.
+  For instance, do not mix cosmetical changes with other, more relevant ones.
+
+- Commit frequently and in an incremental manner.
+  Do not be afraid of making too many commits, they can be merged afterwards.
```

Comment:
Careful, "merge" has a very specific meaning in git, and not the one you are trying to convey here: "squash".

Reply:
Yup, you are right. I was actually thinking that maybe it would be more descriptive for people that are not familiar with git, but it cannot be used like this.

Reply:
I understand completely where you're coming from. How about: "they can be combined afterwards. (This is known as *squashing* in git.)"?

Reply:
Yes, that's nice.